libethereum/Client.cpp

```diff
@@ -82,6 +82,8 @@ Client::~Client()
 {
 	stopWorking();
 	terminate();
+	m_tqReady = {}; // m_tqReady holds a closure that uses m_signalled.
```

Comment:
Not sure, how can this help anything? It is a `shared_ptr<function>`, and you reset it here in `Client`, but `TransactionQueue` still will hold a non-null shared_ptr to a function calling `Client::onTransactionQueueReady()`

Reply:
`m_tqReady` is a member of `ClientBase`.  After `Client` is destroyed but `ClientBase` is alive, when `m_tqReady()` is called, it tries to access a member of `Client` that has been destroyed.

Reply:
And I don't see any TSAN warnings without these.  This was cargo cult.

Reply:
About the `TransactionQueue` problem I filed an issue https://github.com/ethereum/cpp-ethereum/issues/4608